src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -271,6 +276,68 @@ private static ImmutableArray<Diagnostic> EmitRegexMethod(IndentedTextWriter wri
                 writer.WriteLine($"            }}");
             }
 
+            if ((requiredHelpers & RequiredHelperFunctions.IsBoundary) != 0)
+            {
+                writer.WriteLine();
+                writer.WriteLine($"            /// <summary>Determines whether the character at the specified index is a boundary.</summary>");
+                writer.WriteLine($"            [global::System.Runtime.CompilerServices.MethodImpl(global::System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]");
+                writer.WriteLine($"            private static bool IsBoundary(global::System.ReadOnlySpan<char>inputSpan, int index, int startpos, int endpos)");
+                writer.WriteLine($"            {{");
+                writer.WriteLine($"                return (index > startpos && IsBoundaryWordChar(inputSpan![index - 1])) !=");
+                writer.WriteLine($"                       (index < endpos && IsBoundaryWordChar(inputSpan![index]));");
+                writer.WriteLine();
+                writer.WriteLine($"                bool IsBoundaryWordChar(char ch)");
```

Comment:
At the expense of another comparison or two, we could also save all this code and emit something simpler, like:
```C#
bool IsBoundaryWordChar(char ch) =>
    IsWordChar(ch) ||
    (ch == ZeroWidthJoiner | ch == ZeroWidthNonJoiner);
```